containers/ecr-viewer/src/middlewares/withNbsAuth.ts

```diff
@@ -15,12 +15,15 @@ export const withNbsAuth: MiddlewareFactory = (next: ChainableMiddleware) => {
     const nbsAuthResp = set_auth_cookie(request);
     if (nbsAuthResp) return nbsAuthResp;
 
-    if (await authorize_api(request)) return next(request);
+    // NBS auth can only be used for ecr viewer pages
+    const { pathname } = request.nextUrl;
+    if (!pathname.endsWith(`/view-data`)) return next(request);
 
-    return NextResponse.rewrite(
-      new URL(`${process.env.BASE_PATH}/error/auth`, request.nextUrl.origin),
-      { request },
-    );
+    const authorized = await authorize_api(request);
```

Comment:
🤏 nit. I think this function `authorize_api` is misleading, can we rename it to be `isAuthenticated`/`isAuthorized`?

Reply:
I went with `checkIsAuthorized` (I feel like NBS auth does both authn and authz for these pages, so went with the latter)